lib/recipes-data/src/lib/extract_recipedata_from_element.test.ts

```diff
@@ -0,0 +1,375 @@
+          }
+        },
+        {
+          type: ElementType.IMAGE,
+          assets: [
+            {
+              type: AssetType.IMAGE,
+              mimeType: "image/jpeg",
+              file: "https://media.guim.co.uk/58c32a98ae4463b5129bf717b1b2312d8ffc0d45/0_318_5512_3308/1000.jpg",
+              typeData: {
+                aspectRatio: "5:3",
+                width: 1000,
+                height: 600
+              }
+            },
+            {
+              type: AssetType.IMAGE,
+              mimeType: "image/jpeg",
+              file: "https://media.guim.co.uk/58c32a98ae4463b5129bf717b1b2312d8ffc0d45/0_318_5512_3308/500.jpg",
+              typeData: {
+                aspectRatio: "5:3",
+                width: 500,
+                height: 300
+              }
+            },
+            {
+              type: AssetType.IMAGE,
+              mimeType: "image/jpeg",
+              file: "http://media.guim.co.uk/58c32a98ae4463b5129bf717b1b2312d8ffc0d45/0_318_5512_3308/master/5512.jpg",
+              typeData: {
+                aspectRatio: "5:3",
+                width: 5512,
+                height: 3308,
+                isMaster: true
+              }
+            }
+          ],
+          imageTypeData: {
+            caption: "Thomasina Miers’ soba noodles with crisp rainbow vegetables and a spicy sesame seed dressing.",
+            copyright: "LOUISE HAGGER",
+            displayCredit: true,
+            credit: "Photograph: Louise Hagger for the Guardian. Food styling: Emily Kydd. Prop styling: Jennifer Kay",
+            source: "Louise Hagger for the Guardian. Food styling: Emily Kydd. Prop styling: Jennifer Kay",
+            alt: "Thomasina Miers’ soba noodles with crisp rainbow vegetables and a spicy sesame seed dressing.",
+            mediaId: "58c32a98ae4463b5129bf717b1b2312d8ffc0d45",
+            mediaApiUri: "https://api.media.gutools.co.uk/images/58c32a98ae4463b5129bf717b1b2312d8ffc0d45",
+            suppliersReference: "Soba noodles with crisp rainbow vegetables and a spicy sesame se",
+            imageType: "Photograph"
+          }
+        }
+      ]
+    }
+    const result = extractRecipeData(webUrl, block)
+    expect(result.length).toEqual(3)
+    expect(result[2].recipeUId).toEqual("http://www.code.dev-theguardian.com/lifeandstyle/2018/jan/05/soba-noodle-salad-vegetables-spicy-sesame-dressing-recipe-thomasina-miers-3") //
```

Comment:
The ID should not have the full-length URL in it - just the bit from `lifeandstyle/....`.  (mentioned in above comment)